packages/sync/legacy/class.jetpack-sync-actions.php

```diff
@@ -113,7 +114,9 @@ static function sync_allowed() {
 		if ( Jetpack::is_staging_site() ) {
 			return false;
 		}
-		if ( ! Jetpack::is_active() ) {
+
+		$connection_manager = Jetpack_Sync_Main::init()->get_connection_manager();
```

Comment:
Rather than exposing the entire connection here, this condition (and all the others above it) could just be moved into the Jetpack_Sync_Main class.

This prevents us from leaking the connection object outside the class.